tests/cloud_test_logic/cloud_test_index.py

```diff
@@ -6,65 +6,112 @@ class CloudTestIndex(str, Enum):
-        'index_defaults': {
-            'treat_urls_and_pointers_as_images': False,
-            'model': 'test-model',
-            'model_properties': {
-                'name': 'sentence-transformers/multi-qa-MiniLM-L6-cos-v1',
-                'dimensions': 384,
-                'tokens': 128,
-                'type': 'sbert'
-            },
-            'normalize_embeddings': True,
-            "text_preprocessing": {
-                "split_length": 2,
-                "split_overlap": 1,
-            },
-        },
-        "inference_type": "marqo.CPU.large", "storage_class": "marqo.basic",
+    CloudTestIndex.unstructured_basic_index: {
+        "type": "unstructured",
+        "model": "sentence-transformers/all-MiniLM-L6-v2",
+        "inferenceType": "marqo.CPU.large",
+        "storageClass": "marqo.basic"
     },
-    CloudTestIndex.image_index_with_preprocessing_method: {
-        "index_defaults": {
-            "treat_urls_and_pointers_as_images": True,
-            "model": "ViT-B/16",
-            "image_preprocessing": {
-                "patch_method": "simple"
-            },
-        },
-        "inference_type": "marqo.CPU.large", "storage_class": "marqo.basic",
+    # CloudTestIndex.unstructured_image_index: {
+    #     "type": "unstructured",
+    #     "treatUrlsAndPointersAsImages": True,
+    #     "model": "open_clip/ViT-B-32/laion400m_e32",
+    #     "inferenceType": "marqo.CPU.large",
+    #     "storageClass": "marqo.basic"
+    # },
+    # CloudTestIndex.unstructured_text_index_with_custom_model: {
+    #     "type": "unstructured",
+    #     "treatUrlsAndPointersAsImages": False,
+    #     "model": "test-model",
+    #     "modelProperties": {
+    #         "name": "sentence-transformers/multi-qa-MiniLM-L6-cos-v1",
+    #         "dimensions": 384,
+    #         "tokens": 128,
+    #         "type": "sbert"
+    #     },
+    #     "normalizeEmbeddings": True,
+    #     "textPreprocessing": {
+    #         "splitLength": 2,
+    #         "splitOverlap": 1,
+    #         "splitMethod": "sentence",
+    #     },
+    #     "inferenceType": "marqo.CPU.large",
+    #     "storageClass": "marqo.basic"
+    # },
+    # CloudTestIndex.unstructured_image_index_with_preprocessing_method: {
+    #     "type": "unstructured",
+    #     "treatUrlsAndPointersAsImages": True,
+    #     "model": "open_clip/ViT-B-16/laion400m_e31",
+    #     "imagePreprocessing": {
+    #         "patchMethod": "simple"
+    #     },
+    #     "inferenceType": "marqo.CPU.large",
+    #     "storageClass": "marqo.basic"
+    # },
+    #Structured indexes
+    CloudTestIndex.structured_basic_index: {
+        "type": "structured",
+        "inferenceType": "marqo.CPU.large",
+        "model": "sentence-transformers/all-MiniLM-L6-v2",
+        "allFields": [{"name": "text_field_1", "type": "text"}],
+        "tensorFields": ["text_field_1"],
+        "storageClass": "marqo.basic"
     },
```

Comment:
And at least one index should have a dash (`-`) in its name and one should have an underscore (`_`) - just to verify this all works.

Reply:
also. there should be a test where two indexes are the same, but one uses a hyphen and the other uses a dash. For example: 
`my-index`, `my_index`. This is because the index name part of the index's URL will have substitute dashes for underscores (and a random string for uniqueness) - but we should include it as a test to ensure that it works

Reply:
All the comments are addressed except those related to `dash`